## Listas 

Crear una lista 

In [1]:
planets = ['Mercury', 'Venus', 'Earth', 'Mars', 'Jupiter', 'Saturn', 'Uranus', 'Neptune']

acceder a lista por indice 

In [2]:
print('The first planet is', planets[0])
print('The second planet is', planets[1])
print('The third planet is', planets[2])

The first planet is Mercury
The second planet is Venus
The third planet is Earth


modificar valores en una lista 

In [3]:
planets[3] = 'Red Planet'
print('Mars is also known as', planets[3])

Mars is also known as Red Planet


longitud lista 

In [4]:
number_of_planets = len(planets)
print('There are', number_of_planets, 'planets in the solar system.')

There are 8 planets in the solar system.


listas dinamicas 

In [5]:
planets.append('Pluto')
number_of_planets = len(planets)
print('There are actually', number_of_planets, 'planets in the solar system.')

There are actually 9 planets in the solar system.


elimanar valores de listas 

In [6]:
planets.pop()  # Goodbye, Pluto
number_of_planets = len(planets)
print('No, there are definitely', number_of_planets, 'planets in the solar system.')

No, there are definitely 8 planets in the solar system.


indice negativo 

In [7]:
print('The last planet is', planets[-1])
print('The penultimate planet is', planets[-2])

The last planet is Neptune
The penultimate planet is Uranus


busacr valor en una lista 

In [8]:
jupiter_index = planets.index('Jupiter')
print('Jupiter is the', jupiter_index + 1, 'planet from the sun')

Jupiter is the 5 planet from the sun


almacenar numeros en listas 

In [9]:
gravity_on_earth = 1.0
gravity_on_the_moon = 0.166

In [10]:
gravity_on_planets = [0.378, 0.907, 1, 0.379, 2.36, 0.916, 0.889, 1.12]

In [11]:
bus_weight = 12650 # in kilograms, on Earth

print('On Earth, a double-decker bus weighs', bus_weight, 'kg')
print('On Mercury, a double-decker bus weighs', bus_weight * gravity_on_planets[0], 'kg')

On Earth, a double-decker bus weighs 12650 kg
On Mercury, a double-decker bus weighs 4781.7 kg


debera salir: 
Salida: On Earth, a double-decker bus weighs 12650 kg
Salida: On Mercury, a double-decker bus weighs 4781.7 kg

In [12]:
bus_weight = 12650 # in kilograms, on Earth

print('On Earth, a double-decker bus weighs', bus_weight, 'kg')
print('The lightest a bus would be in the solar system is', bus_weight * min(gravity_on_planets), 'kg')
print('The heaviest a bus would be in the solar system is', bus_weight * max(gravity_on_planets), 'kg')

On Earth, a double-decker bus weighs 12650 kg
The lightest a bus would be in the solar system is 4781.7 kg
The heaviest a bus would be in the solar system is 29854.0 kg


debera salir : 
Salida:
* On Earth, a double-decker bus weighs 12650 kg
* The lightest a bus would be in the solar system is 4781.7 kg
* The heaviest a bus would be in the solar system is 29854 kg

manipulacion de datos de lista 

In [13]:
planets = ['Mercury', 'Venus', 'Earth', 'Mars', 'Jupiter', 'Saturn', 'Uranus', 'Neptune']
planets_before_earth = planets[0:2]
print(planets_before_earth)

['Mercury', 'Venus']


Observa cómo la Tierra no está incluida en la lista. La razón es que el índice termina antes que el índice final.

Para obtener todos los planetas después de la Tierra, comenzamos en el tercero y vamos al octavo:

In [14]:
planets_after_earth = planets[3:8]
print(planets_after_earth) 

['Mars', 'Jupiter', 'Saturn', 'Uranus', 'Neptune']


unir listas 

In [15]:
amalthea_group = ['Metis', 'Adrastea', 'Amalthea', 'Thebe']
galilean_moons = ['Io', 'Europa', 'Ganymede', 'Callisto']

regular_satellite_moons = amalthea_group + galilean_moons
print('The regular satellite moons of Jupiter are', regular_satellite_moons)


The regular satellite moons of Jupiter are ['Metis', 'Adrastea', 'Amalthea', 'Thebe', 'Io', 'Europa', 'Ganymede', 'Callisto']


Ordenar listas 

In [16]:
regular_satellite_moons.sort()
print("The regular satellite moons of Jupiter are", regular_satellite_moons)

The regular satellite moons of Jupiter are ['Adrastea', 'Amalthea', 'Callisto', 'Europa', 'Ganymede', 'Io', 'Metis', 'Thebe']


In [17]:
regular_satellite_moons.sort(reverse=True)
print("The regular satellite moons of Jupiter are", regular_satellite_moons)

The regular satellite moons of Jupiter are ['Thebe', 'Metis', 'Io', 'Ganymede', 'Europa', 'Callisto', 'Amalthea', 'Adrastea']


## Practicas 1 Kata

Lista de pklanetas
En primer lugar, crea una variable denominada planets. Agrega los ocho planetas (sin Plutón) a la lista. A continuación, muestra el número de planetas.

In [20]:
## Primero creamos lista de planetas 

planetas = ["Mercurio", "Venus", "Tierra", "Marte", "Jupiter", "Saturno", "Urano", "Nepturno"]

print("Son", len(planetas), "planetas")

Son 8 planetas


Agrega a Plutón a la lista que creaste. Luego muestra tanto el número de planetas como el último planeta de la lista.

In [22]:
planetas.append("Pluton")
print(planetas[-1], "es el noveno planeta")

Pluton es el noveno planeta


## Ejercicio 2 kata 

Es posible que debas trabajar con diferentes secciones de una lista. En nuestro ejemplo, queremos mostrar planetas más cerca y más lejos del sol de un planeta que el usuario ingresa por teclado.

Comienza agregando el código para crear una lista con los planetas.

In [ ]:
planetas = ["Mercurio", "Venus", "Tierra", "Marte", "Jupiter", "Saturno", "Urano", "Nepturno"]

A continuación, agrega el código para solicitar al usuario un nombre. Debido a que las cadenas distinguen entre mayúsculas y minúsculas en Python, pídale al usuario que use una letra mayúscula para comenzar el nombre del planeta.

In [24]:
ing_plan = input("Ingrese el nombre de un plenta comenzando con mayuscula")

KeyboardInterrupt: Interrupted by user

Para determinar qué planetas están más cerca que el que ingresó el usuario, debes encontrar dónde está el planeta en la lista. Puedes utilizar index para realizar esta operación. Agrega el código para encontrar el índice del planeta.

In [25]:
plan_index = planetas.index(ing_plan)

NameError: name 'ing_plan' is not defined

Con el índice determinado, ahora puedes agregar el código para mostrar los planetas más cercanos al sol.


In [ ]:
print("El planeta mas cerca es: /n " + ing_plan)
print(planetas[0:ing_plan])

Puedes usar el mismo índice para mostrar planetas más alejados del sol. Sin embargo, recuerda que el índice inicial se incluye cuando usas un slice. Como resultado, tendrás que agregar 1 al valor. Agrega el código para mostrar los planetas más alejados del sol.


In [ ]:
print("El planetas mas cercano es: /n" + ing_plan)
print(planetas[plan_index + 1:])